Update PckgCI regression test to use new special_models test suite.

saienduri opened the pull request:
This commit updates the SHARK-TestSuite ref and makes the neccessary changes to adhere to the new special models test suite. With this, we move away from using all the specialized json config files and complex workflows.
Instead, we use python scripts which allow us to use custom flags, tolerances, and configurations based on the backend/model.
Related PR in TestSuite: https://github.com/nod-ai/SHARK-TestSuite/pull/271

ci-exactly: build_all, test_amd_mi300, build_packages, regression_test

saienduri committed:
updates to sync with latest test suite

Signed-off-by: saienduri <saimanas.enduri@amd.com>

saienduri committed:
pre-commit and see if iree's ireers install is interfering with shark-test-suite ireers

Signed-off-by: saienduri <saimanas.enduri@amd.com>

saienduri committed:
add common_tools installation for onnx as well and try different TEST_SUITE_REPO_ROOT path

Signed-off-by: saienduri <saimanas.enduri@amd.com>

saienduri committed:
remove pip freeze debug commands

Signed-off-by: saienduri <saimanas.enduri@amd.com>

ScottTodd commented on .github/workflows/pkgci_regression_test.yml:
```diff
@@ -90,14 +90,16 @@ jobs:
       - name: Install external TestSuite Python requirements
         run: |
           source ${VENV_DIR}/bin/activate
+          pip uninstall -y iree-regression-suite
```
This is a code smell - installing a package then uninstalling it.

For an incremental step, we could teach `setup_venv.py` to optionally install `iree-regression-suite` and have this job skip that part: https://github.com/iree-org/iree/blob/5f2ebe99c62e1259ec9e50d1aaed520c3f66b5f6/build_tools/pkgci/setup_venv.py#L121-L129

I think what we should ultimately do (like, within 1-2 weeks) is merge (either direction) the existing `experimental/regression_suite` with `SHARK-TestSuite/iree_special_models`

We should also be aiming to move code out of experimental. The new tests here look similar to the rest of `experimental/regression_suite` but use other packages. That's a confusing state to be in. If the new code is also experimental but shares no code directly then it could go in a sibling directory.

ScottTodd commented on experimental/regression_suite/shark-test-suite-models/sd3/test_clip.py:
```diff
@@ -0,0 +1,108 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
```
Use IREE copyright in this repo.

```suggestion
# Copyright 2024 The IREE Authors
```

ScottTodd commented on experimental/regression_suite/shark-test-suite-models/sd3/test_clip.py:
```diff
@@ -0,0 +1,108 @@
+import os
+
+repo_root = os.getenv("TEST_SUITE_REPO_ROOT")
+current_dir = repo_root + "/iree_special_models/sd3/prompt-encoder"
```
I'm still on the fence about how best to architect this, but my gut says this is still inverted and awkward to use.

The `experimental/regression_suite` code (which is stale / unmaintained) uses code like
```python
llama2_7b_f16qi4_stripped_source = fetch_source_fixture(
    "https://sharktank.blob.core.windows.net/sharktank/llama_regression/09152023/llama2_7b_int4_stripped.mlir",
    group="llama2_7b_f16qi4_stripped",
)
```
that fetches as part of the test and doesn't rely on a git repo existing somewhere on disk.

similarly, some new sharktank tests (https://github.com/nod-ai/sharktank/blob/main/sharktank/integration/models/punet/integration_test.py) do this:
```python
@pytest.fixture(scope="module")
def sdxl_fp16_base_files():
    from huggingface_hub import hf_hub_download

    REPO_ID = "stabilityai/stable-diffusion-xl-base-1.0"
    REVISION = "76d28af79639c28a79fa5c6c6468febd3490a37e"

    def download(filename):
        return hf_hub_download(
            repo_id=REPO_ID, subfolder="unet", filename=filename, revision=REVISION
        )

    return {
        "config.json": download("config.json"),
        "params.safetensors": download("diffusion_pytorch_model.fp16.safetensors"),
    }
```

(Stella wrote both of those tests and it's a style I like but am still getting used to)

I'd prefer that style of test file fetching here over the `download_remote_files.py` script, so developers can run `pytest` directly without any other repo fetching: https://github.com/iree-org/iree/tree/main/experimental/regression_suite#quick-start

The main reasons to put files outside of the core IREE repo are:
* Keeping the core repo focused (lower commit volume, lower complexity, more shared between backends / contributors)
* Sharing files across multiple projects (e.g. out of tree plugins like https://github.com/nod-ai/iree-amd-aie)

ScottTodd commented on .github/workflows/pkgci_regression_test.yml:
```diff
@@ -237,61 +235,33 @@ jobs:
           source ${VENV_DIR}/bin/activate
-          pytest SHARK-TestSuite/iree_tests/pytorch/models \
+          pytest ./experimental/regression_suite/shark-test-suite-models/sdxl \
+            -k ${{ matrix.backend }} \
```
The `amdgpu_vulkan` matrix entry doesn't specify `backend`, so this is failing with `pytest: error: argument -k: expected one argument`

https://github.com/iree-org/iree/actions/runs/9779308546/job/27006098126?pr=17803#step:11:28

ScottTodd reviewed (changes requested):
Hmmm... I'm feeling like the best path forward here is keep all the "special model" code in-tree (not relying on anything in SHARK-TestSuite) and just extend the existing `experimental/regression_suite` tests. The ability to just run `pytest` and have files downloaded automatically is really useful.

IDK. Probably need to take a step back and put a plan together for "core" project tests, community tests, etc. We're accumulating quite a bit of tech debt by piling new tests on top of old tests without first updating the old tests.

saienduri committed:
try adding mi300

Signed-off-by: saienduri <saimanas.enduri@amd.com>

saienduri committed:
add workflow changes for test_amd_mi300 as well

Signed-off-by: saienduri <saimanas.enduri@amd.com>

saienduri committed:
add rocm-chip

Signed-off-by: saienduri <saimanas.enduri@amd.com>

saienduri committed:
rocm chip for benchmark

Signed-off-by: saienduri <saimanas.enduri@amd.com>

ScottTodd committed:
Simplify tests/e2e/linalg_ext_ops. (#17856)

Forked from https://github.com/iree-org/iree/pull/17766

* Share test srcs lists between:
  * Vulkan and Metal (both using SPIR-V codegen)
  * CUDA and ROCm (both using LLVMGPU codegen)
* Enable `winograd_input.mlir` and `winograd_output.mlir` tests on more
backends
* Add Metal and ROCm/HIP tests
* Skip wasm tests using a label instead of CMake branching

ScottTodd committed:
Disable failing Pixel6/Vulkan linalg_ext reverse test again. (#17861)

See https://github.com/iree-org/iree/pull/17856#discussion_r1673193963

qedawkins committed:
Drop unmaintained transform dialect tests (#17858)

The tests in `tests/transform_dialect` have been unmaintained (and
mostly disabled) for a while. Most of the tests were misplaced anyway,
mixing lit tests in with e2e tests, and much of the relevant
functionality is now tested upstream or has evolved into different lit
tests in Codegen.

This drops the entire `cuda` directory, and drops all `cpu` tests except
one which is testing a transform dialect library call, which is still a
relevant e2e test.

ScottTodd committed:
Deflake ScopeTest.WaitIdleFailure by increasing sleep time. (#17859)

This test has flaked a few times:

*
https://github.com/iree-org/iree/actions/runs/9653016869/job/26624436852#step:5:1882
*
https://github.com/iree-org/iree/actions/runs/9716312176/job/26819673196#step:4:18786

```
[ RUN      ] ScopeTest.WaitIdleFailure
/work/runtime/src/iree/task/scope_test.cc:225: Failure
Value of: iree_task_scope_is_idle(&scope)
  Actual: true
Expected: false

[  FAILED  ] ScopeTest.WaitIdleFailure (175 ms)
[----------] 9 tests from ScopeTest (326 ms total)

[----------] Global test environment tear-down
[==========] 9 tests from 1 test suite ran. (326 ms total)
[  PASSED  ] 8 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ScopeTest.WaitIdleFailure

 1 FAILED TEST
```

We could instead drop the
`EXPECT_FALSE(iree_task_scope_is_idle(&scope));` check entirely.
Increasing the sleep time will increase test time regardless of OS
scheduler behavior (unlike https://github.com/iree-org/iree/pull/17857
which increased a _timeout_ duration).

ScottTodd committed:
Deflake LoopTest.WaitOneBlocking by increasing timeout. (#17857)

This test case has failed a few times on macos_x86, and possibly on
other platforms too. I suspect the OS isn't scheduling the test threads
soon enough - there's only a 150 millisecond window between the event
set on a worker thread and `iree_loop_wait_one`'s timeout on the main
thread:

```
main thread                worker thread
-----------                -------------
test start
spin up thread   ------>   wait 50ms
wait 200ms                 ...
...              <------   set event
timeout if event not set
```

Sample logs:
*
https://github.com/iree-org/iree/actions/runs/9214985535/job/25352380335#step:10:1578
*
https://github.com/iree-org/iree/actions/runs/9882364677/job/27295096340?pr=17856#step:9:43

```
[ RUN      ] LoopTest.WaitOneBlocking
iree/runtime/src/iree/base/loop_test.h:612: Failure
Value of: status
Expected: error code OK
  Actual: 0x4, whose error code is DEADLINE_EXCEEDED: DEADLINE_EXCEEDED

[  FAILED  ] LoopTest.WaitOneBlocking (200 ms)
```

We could also use an infinite timeout.

ScottTodd committed:
Bump goldentime values for mi250 sdxl benchmarks. (#17860)

Seeing some variance causing CI failures with unrelated changes:
*
https://github.com/iree-org/iree/actions/runs/9879301822/job/27286342282#step:15:46
  * `VAE Decode Benchmark Time: 295.0 ms (golden time 291.0 ms)`
  * `Prompt Encoder Benchmark Time: 17.2 ms (golden time 17.0 ms)`
*
https://github.com/iree-org/iree/actions/runs/9877397187/job/27279465007
  * `VAE Decode Benchmark Time: 295.0 ms (golden time 291.0 ms)`
  * `Prompt Encoder Benchmark Time: 17.1 ms (golden time 17.0 ms)`

Aside: we might want to move the benchmarks to a `pkgci_benchmarks.yml`
workflow instead of chaining them on to the end of
`pkgci_regression_test.yml`.

ci-exactly: build_packages,regression_test

ScottTodd committed:
Deflake LoopTest.WaitAnyBlocking and WaitAllBlocking. (#17863)

Missed these in https://github.com/iree-org/iree/pull/17857 since I only
saw `WaitOneBlocking` flake on CI. The other test cases can flake too.

benvanik committed:
Adding some CTS variants for indirect command buffers. (#17846)

This is a simple start that tests a few commands to test the
infrastructure for record/replay of indirect command buffers and the
validation mechanism. We don't have a good way in the CTS yet to
conditionally run tests based on whether validation is compiled into the
build but I'll be looking into that for testing failure cases in future
PRs (for now I've just tested manually to verify errors are propagated).

krzysz00 committed:
[Codegen] Improve ROCm-specific LLVM translations (#17742)

Use upstream's translations for attributes like rocdl.kernel to reduce
redundancy.

Fix the parsing of chipset versions (the last two digits are in base 16)

Signed-off-by: Krzysztof Drewniak <Krzysztof.Drewniak@amd.com>

hanhanW committed:
[GPU] Fix the propagation control function logic. (#17869)

This is a follow-up from the integrate comment:
https://github.com/iree-org/iree/pull/17827#discussion_r1671197583

The comment was not addressed in the integrate PR because the CI was not
stable. Some runners were off, and we decided to land the PR and sent a
follow-up later. See
https://discord.com/channels/689900678990135345/1080178290188374049/1260330876651307090
for more details.

raikonenfnu committed:
[LinalgExt] Adding IndexingMaps to linalg_ext.attentionOp (#17864)

In order to make fusion with other generics, specifically transpose
easier, we introduce affineMaps/indexingMaps to linalg_ext.attentionOp.
With that we are also enforcing the number and types of dpsInputs. We
are also removing "transpose_V" attribute in lieu of infering from
indexingMaps.

---------

Co-authored-by: Kunwar Grover <groverkss@gmail.com>

ScottTodd committed:
Bump goldensize values for sdxl benchmarks. (#17873)

ci-exactly: build_packages,regression_test

ScottTodd committed:
Round up sdxl golden dispatch sizes and mi250 times by 10%. (#17879)

See [this discussion on
Discord](https://discord.com/channels/689900678990135345/689957613152239638/1261059802214305913).

These are change detector tests with _some_ use, but exact checking is
too noisy and is causing significant churn across PRs and postsubmit CI
runs. We could change the test script to allow a certain error range,
but for now just increase the thresholds used in `<=` checks by about
10% (then drop a few digits to 0 so we don't give the appearance of high
precision).

ci-exactly: build_packages,regression_test

saienduri committed:
test out all changes for regression testing, no benchmarking changes yet

Signed-off-by: saienduri <saimanas.enduri@amd.com>

saienduri committed:
pre-commit and formatting stuff

Signed-off-by: saienduri <saimanas.enduri@amd.com>
